lib/std/hash_map.zig

```diff
@@ -1505,6 +1511,85 @@ pub fn HashMapUnmanaged(
             return result;
         }
 
+        /// Rehash the map, in-place
```

Comment:
what does it do?

when should you call this?

what happens to existing key/value pointers?

all this information should be in the doc comments

Reply:
How would code know when there is "excessive" tombstone buildup (not by performance monitoring)? Can you query for the stats?

Reply:
I don't think you can easily determine that, in my original issue on #17851 it took a somewhat larger number of iterations to expose the performance issue.

This is a band-aid and the best fix would be a different HashMap implementation, but without this patch there are scenarios where the HashMap has terrible performance by default.

Reply:
> When you say doc comments, do you mean `///` comments before the `pub fn rehash` line? Or somewhere else?

Yes, precisely - triple slash comments in front of the function.

Reply:
That's great.

What do you think of 73df0b194f7de1099d93807170f9d5207d2ab431